includes/promoted-jobs/class-wp-job-manager-promoted-jobs.php

```diff
@@ -161,6 +162,10 @@ public static function update_promotion( $post_id, $promoted ) {
 
 		delete_option( self::PROMOTED_JOB_TRACK_OPTION );
 
+		if ( 'force_delete' === $promoted ) {
```

Comment:
I'm thinking here about this. It is a bit weird for me to have a parameter that might be a boolean, but also might be a string, and a string with a special value (otherwise, it is interpreted as `true`).

I'm not sure what the best approach here:

1. Add a third parameter in the spirit of `$force_delete` on `wp_delete_post`, that, when `$promoted` is `false` and that parameter is `true` (the default value is `false`), it deletes the post meta as appropriate
2. Maybe copy some of the logic present here (to validate if the post type is correct and delete the promoted job track option) to a private method , and implement the change directly on `deactivate_promotion`.

Also, now that I'm thinking here. This change that I suggested basically implies that, after a job is expired on WPJMCOM, it will never delete the post meta here on WPJM, right? It will just set `_promoted` to `0` and call it a day. :thinking:

Reply:
> I'm thinking here about this. It is a bit weird for me to have a parameter that might be a boolean, but also might be a string, and a string with a special value (otherwise, it is interpreted as true).

Implemented your suggestion 2 here: https://github.com/Automattic/WP-Job-Manager/pull/2562/commits/f0859e7663272fcbdd8b6bdb807f8feba23c2500

> Also, now that I'm thinking here. This change that I suggested basically implies that, after a job is expired on WPJMCOM, it will never delete the post meta here on WPJM, right? It will just set _promoted to 0 and call it a day. 🤔

Yep! This was the way to differentiate the cases where the user is deactivating manually (that we really want to remove from the feed in order to deactivate the promotions). Otherwise, the edge case of accidentally stopping a promotion because of the interval sync could happen. Pasting here the case, just for context:

> 1. User promote a job, completing the JobTarget checkout.
> 2. Before the 5 minutes polling interval (or more if they delay to complete the checkout), the user change something in a promoted jobs triggering a sync (WPJM -> [wpjobmanager.com](http://wpjobmanager.com/)).
> 3. Their job would still have _promoted 0 because it didn't sync with the new status yet ([wpjobmanager.com](http://wpjobmanager.com/) -> WPJM). So in the sync WPJM -> [wpjobmanager.com](http://wpjobmanager.com/), it would remove the job from [wpjobmanager.com](http://wpjobmanager.com/), removing from the feed that JobTarget will consume, which would deactivate their promotion.